src/app/DomainPermissionsPage.qml

```diff
@@ -0,0 +1,290 @@
+    property bool selectMode
+    property bool sortByLastRequested: false
+
+    signal done()
+    signal reload(string selectedDomain)
+
+    BrowserPage {
+        id: domainPermissionsPage
+
+        anchors.fill: parent
+        focus: true
+
+        title: i18n.tr("Domain specific permissions")
+
+        showBackAction: !selectMode
+
+        leadingActions: [
+            Action {
+                objectName: "close"
+                iconName: "close"
+                onTriggered: selectMode = false
+            }
+        ]
+
+        trailingActions: [
+            Action {
+                text: i18n.tr("Select all")
+                iconName: "select"
+                visible: selectMode
+                onTriggered: {
+                    if (domainPermissionsListView.ViewItems.selectedIndices.length === domainPermissionsListView.count) {
+                        domainPermissionsListView.ViewItems.selectedIndices = []
+                    } else {
+                        var indices = []
+                        for (var i = 0; i < domainPermissionsListView.count; ++i) {
+                            indices.push(i)
+                        }
+                        domainPermissionsListView.ViewItems.selectedIndices = indices
+                    }
+                }
+            },
+            Action {
+                text: i18n.tr("Delete")
+                iconName: "delete"
+                visible: selectMode
+                enabled: domainPermissionsListView.ViewItems.selectedIndices.length > 0
+                onTriggered: {
+                    var toDelete = []
+                    for (var index = 0; index < domainPermissionsListView.ViewItems.selectedIndices.length; index++) {
+                        var selectedDomainSetting = domainPermissionsListView.model.get(domainPermissionsListView.ViewItems.selectedIndices[index])
+                        toDelete.push(selectedDomainSetting.domain)
+                    }
+                    for (var i = 0; i < toDelete.length; i++) {
+                        DomainPermissionsModel.removeEntry(toDelete[i])
+                    }
+                    domainPermissionsListView.ViewItems.selectedIndices = []
+                    selectMode = false
+                }
+            },
+            Action {
+                iconName: "edit"
+                visible: !selectMode
+                enabled: domainPermissionsListView.count > 0
+                onTriggered: {
+                    selectMode = true
+                }
+            },
+            Action {
+                iconName: "add"
+                visible: !selectMode
+
+                onTriggered: {
+                    var promptDialog = PopupUtils.open(Qt.resolvedUrl("PromptDialog.qml"), domainPermissionsPage);
+                    promptDialog.title = i18n.tr("Add domain")
+                    promptDialog.message = i18n.tr("Enter the name of the domain, e.g. example.com (subdomains will be removed).")
```

Comment:
May this dialog also have a "Current page" option, to add domain settings for the site you're currently viewing?

Reply:
sounds good, I think the settings page cannot directly access this information though. (we might need to add another property to the settings page)
for the zoom, location and custom url schemes there is already a workflow that adds the current page as needed. (e.g. if user changes the zoom or the page needs location access)

Reply:
this feature I would rather skip for now, as soon as custom user agents will (hopefully) be activated again, it will become handy to have it.
For now the location / custom schemes / zoom do already have a workflow to add the current page automatically.

Reply:
Ah, don't worry about it for now then.